modules/configuration/src/test/java/org/apache/ignite/internal/configuration/tree/NamedListNodeTest.java

```diff
@@ -347,4 +348,36 @@ public void errors() throws Exception {
         b.delete("X");
         b.delete("Y");
     }
+
+    /**
+     * Tests the {@link NamedListNode#update} method.
+     */
+    @Test
+    public void testUpdate() {
+        var list = new NamedListNode<SecondChange>("name", () -> cgen.instantiateNode(SecondConfigurationSchema.class), null);
+
+        list.create("foo", ch -> ch.changeStr("bar"));
+
+        assertThat(list.get("foo").str(), is(equalTo("bar")));
+
+        list.update("foo", ch -> ch.changeStr("baz"));
+
+        assertThat(list.get("foo").str(), is(equalTo("baz")));
+
+        list.delete("foo");
+
+        // updating a removed key should throw
+        assertThrows(IllegalArgumentException.class, () -> list.update("foo", ch -> {}));
```

Comment:
Add and run:
        assertThrows(IllegalArgumentException.class, () -> list.create("foo", ch -> ch.changeStr("bar2")));

        list.createAfter("foo", "foo2", ch -> ch.changeStr("a"));
named list allows to add record after deleted one. Is it ok?

Reply:
I've created a ticket: https://issues.apache.org/jira/browse/IGNITE-16042